src/ja/user/config/query.py

```diff
@@ -12,70 +13,297 @@ class QueryCommandConfig(UserConfig):
+    def verbosity(self) -> Verbosity:
+        return self._verbosity
+
+    @property
+    def ssh_config(self) -> SSHConfig:
+        return self._ssh_config
+
+    def __eq__(self, o: object) -> bool:
+        if isinstance(o, QueryCommandConfig):
+            return self._uid == o._uid \
+                   and self._label == o._label \
+                   and self._owner == o._owner \
+                   and self._priority == o._priority \
+                   and self._status == o._status \
+                   and self._is_preemtible == o._is_preemtible \
+                   and self._special_resources == o._special_resources \
+                   and self._cpu_threads == o._cpu_threads \
+                   and self._memory == o._memory \
+                   and self._before == o._before \
+                   and self._after == o._after \
+                   and self._ssh_config == o._ssh_config \
+                   and self._verbosity == o._verbosity
+        else:
+            return False
+
+    def __dir__(self) -> Iterable[str]:
+        return ["_uid", "_label", "_owner", "_priority", "_status", "_is_preemtible",
+                "_special_resources", "_cpu_threads", "_memory", "_before", "_after",
+                "_ssh_config", "_verbosity"]
+
+    def source_from_user_config(self, other_config: "QueryCommandConfig", unset_only: bool = True) -> None:
+        if unset_only is False:
+            self.__init__(other_config._uid, other_config._label, other_config._owner,
+                          other_config._priority, other_config._status, other_config._is_preemtible,
+                          other_config._special_resources, other_config._cpu_threads, other_config._memory,
+                          other_config._before, other_config._after)
+            super().__init__(other_config._ssh_config, other_config._verbosity)
+        else:
+            for attr in dir(self):
+                if getattr(self, attr) is None:
+                    setattr(self, attr, getattr(other_config, attr))
+
+    def to_dict(self) -> Dict[str, object]:
+        _dict: Dict[str, object] = dict()
+        _dict["uid"] = self._uid
+        _dict["label"] = self._label
+        _dict["owner"] = self._owner
+        _dict["priority"] = [a.value for a in self._priority]
+        _dict["status"] = [a.value for a in self._status]
+        _dict["is_preemtible"] = self._is_preemtible
+        _dict["special_resources"] = self._special_resources
+        _dict["cpu_threads"] = self._cpu_threads
+        _dict["memory"] = self._memory
+        _dict["before"] = self._before.strftime("%d/%m/%Y %H:%M:%S")
+        _dict["after"] = self._after.strftime("%d/%m/%Y %H:%M:%S")
+        _dict["ssh_config"] = self._ssh_config.to_dict()
+        _dict["verbosity"] = self._verbosity
+        return _dict
+
+    @classmethod
+    def from_dict(cls, property_dict: Dict[str, object]) -> "QueryCommandConfig":
+        _uid = None
+        if "uid" in property_dict:
```

Comment:
These checks should not be necessary.
cls._get_str_from_dict should return None if the dictionary does not contain the specified property.

Reply:
But it doesn't work with string list.